sdk/storage/src/blobs/block_blob_client.cpp

```diff
@@ -58,11 +58,11 @@ namespace Azure { namespace Storage { namespace Blobs {
   }
 
   BlobContentInfo BlockBlobClient::Upload(
-      std::unique_ptr<Azure::Core::Http::BodyStream> content,
+      Azure::Core::Http::BodyStream& content,
       const UploadBlobOptions& options) const
   {
     BlobRestClient::BlockBlob::UploadOptions protocolLayerOptions;
-    protocolLayerOptions.BodyStream = std::move(content);
+    // protocolLayerOptions.BodyStream = std::move(content);
```

Comment:
We also want to move body content out of options.
BodyStream is a required parameter for HttpRequest, just like url.
So it should not be in operation options.

Confirmed by @jeff  and @RickWinter

Reply:
Blob service changed.

Reply:
I am still in doubt why body stream should be required for **ALL** requests. A request body should never be mandatory even for PUT requests.

Reply:
I agree that request body shouldn't be mandatory for GET/HEAD/DELETE requests. But I think it's mandatory for PUT/POST requests.

Reply:
HTTP allows request bodies for ALL HTTP methods and so the request supports this. Also, the request object doesn't even know the exact method that's being used (it's passed as a parameter). So, since every method CAN have a body and because the body is a separate logical entity of any arbitrary HTTP request the body is an explicitly specified parameter. We added a NullBodyStream which can be passed for HTTP requests that want to have no body.